public/media_engagement/imaqal_media_engagement.js

```diff
@@ -362,5 +305,290 @@ const update = (data) => {
     .attr("text-anchor", "start")
     .style("fill", "blue")
     .text("Total Failed");
+
+
+    function updateView10Minutes() {
+        draw10MinReceivedGraph()
+        draw10MinSentGraph()       
+    }
+
+    function updateViewOneDay() {
+        drawOneDayReceivedGraph()
+        drawOneDaySentGraph()
+    }
+
+    function draw10MinReceivedGraph() {
+
+        var offset = new Date()
+    
+        offset.setDate(offset.getDate() - 7)
+
+        dataFiltered = data.filter(a => a.datetime > offset);
+
+        let stackReceived = d3.stack()
+            .keys(receivedKeys)
+        let receivedDataStacked = stackReceived(dataFiltered)
+
+        // set scale domains
+        x.domain(d3.extent(dataFiltered, d => new Date(d.datetime)));
+        y_total_received_sms.domain([0, d3.max(dataFiltered, function (d) { return d.total_received; })]);
+
+        d3.selectAll("#receivedStack").remove();
```

Comment:
Is there a way of removing all of these in one go? Otherwise we get something that creates a coupling between all fo the addition and all of the removal code. So if there isn't a way we'll need to refactor this into a function.

Reply:
The # is different in each case, refactored the rest using a class. Good enough?